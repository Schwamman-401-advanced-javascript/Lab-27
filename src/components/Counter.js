import React from 'react';

class Counter extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      counter: 0,
    };
  }

  incrementClick = e => {
    e.preventDefault();
    this.setState({ counter: this.state.counter + 1 });
  };

  decrementClick = e => {
    e.preventDefault();
    this.setState({ counter: this.state.counter - 1});
  }

  render() {
    let counterColor = '';

    if (this.state.counter > 0 && this.state.counter <= 5) {
      counterColor = 'lightGreen';
    } else if (this.state.counter > 5 && this.state.counter <= 10) {
      counterColor = 'medLightGreen';
    } else if (this.state.counter > 10 && this.state.counter <= 15) {
      counterColor = 'medDarkGreen';
    } else if (this.state.counter > 15) {
      counterColor = 'darkGreen';
    } else if (this.state.counter < 0 && this.state.counter >= -5) {
      counterColor = 'lightRed';
    } else if (this.state.counter < -5 && this.state.counter >= -10) {
      counterColor = 'medLightRed';
    } else if (this.state.counter < -10 && this.state.counter >= -15) {
      counterColor = 'medDarkRed';
    } else if (this.state.counter < -15) {
      counterColor = 'darkRed';
    } else if(document.getElementById('counter')){
      counterColor = '';
    }
   


    return (
      <div>
        <h4 id="counter" className={counterColor}>{this.state.counter}</h4>
        <button id="increment" onClick={this.incrementClick}>Increment</button>
        <button id="decrement" onClick={this.decrementClick}>Decrement</button>
      </div>
    );
  }
}

export default Counter;