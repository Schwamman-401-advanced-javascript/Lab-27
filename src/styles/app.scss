$background-color: #F2EEC8;

body {
  background-color: $background-color;
  font-family: monospace;
  color: #2F2C32;
}

header {
  background-color: #59875A;
  margin-top: -30px;
  margin-left: -5%;
  padding: 5px;
  height: 145px;
  text-align: center;
  font-size: 2rem;
  width: 110%;
  border-bottom: solid 5px #2F2C32;

}

#root {
  div {
    padding: 5px;
    width: 50%;
    margin: 60px;
    margin-left: auto;
    margin-right: auto;
    background-color: #dbe2ef;
    text-align: center;
    border: dotted 3px black;
    border-radius: 15px;
  }

  h4 {
    font-weight: bold;
    font-size: 3rem;
    width: 30%;
    padding: 15px;
    margin-left: auto;
    margin-right: auto;
    border: solid 3px black;
    border-radius: 5px;
    -webkit-box-shadow: 10px 10px 5px 0px rgba(0,0,0,0.75);
    -moz-box-shadow: 10px 10px 5px 0px rgba(0,0,0,0.75);
    box-shadow: 10px 10px 5px 0px rgba(0,0,0,0.75);
  }

  button {
    padding: 10px;
    margin-left: 10px;
    margin-right: 10px;
    margin-bottom: 10px;
    margin-top: -30px;
    font-size: 1.6rem;
    border-radius: 5px;
    background-color: #B9693A;
    -webkit-box-shadow: 10px 10px 5px 0px rgba(0,0,0,0.75);
    -moz-box-shadow: 10px 10px 5px 0px rgba(0,0,0,0.75);
    box-shadow: 10px 10px 5px 0px rgba(0,0,0,0.75);
  }

}

footer {
  text-align: center;
}

.lightGreen {
      background-color: rgba(113, 225, 76, .25);
    }

    .medLightGreen {
      background-color: rgba(113, 225, 76, .50);
    }

    .medDarkGreen {
      background-color: rgba(113, 225, 76, .75);
    }

    .darkGreen {
      background-color: rgba(113, 225, 76, 1);
    }

    .lightRed {
      background-color: rgba(188, 32, 45, .25);
    }

    .medLightRed {
      background-color: rgba(188, 32, 45, .50);
    }

    .medDarkRed {
      background-color: rgba(188, 32, 45, .75);
    }

    .darkRed {
      background-color: rgba(188, 32, 45, 1);
    }

