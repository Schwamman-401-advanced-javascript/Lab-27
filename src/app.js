import React from 'react';
import Header from './components/Header';
import Footer from './components/Footer';
import Counter from './components/Counter';
import './styles/app.scss';


class App extends React.Component {
  render() {
    return (
      <React.Fragment>
        <Header />
        <Counter />
        <Footer />
      </React.Fragment>
    );
  }
}

export default App;
