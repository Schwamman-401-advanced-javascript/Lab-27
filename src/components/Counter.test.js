import React from 'react';
import app from '../app';
import Counter from './Counter';
import renderer from 'react-test-renderer';

describe('<Counter />', () => {
  it('can render initially', () => {
    let app = shallow(<Counter />);
    expect(app.find('h4#counter').exists()).toBe(true);
  });

  it('has count: 0 initially', () => {
    let app = shallow(<Counter />);
    expect(app.state('counter')).toBe(0);

    let counterHeader = app.find('h4#counter');
    expect(counterHeader.exists()).toBe(true);
    expect(counterHeader.text()).toBe('0');
  });

  it('can increment count on button click', () => {
    let app = mount(<Counter />);
    let button = app.find('button#increment');
    expect(app.state('counter')).toBe(0);

    button.simulate('click');
    expect(app.state('counter')).toBe(1);
    expect(app.find('h4#counter').text()).toBe('1');

    button.simulate('click');
    expect(app.state('counter')).toBe(2);
    expect(app.find('h4#counter').text()).toBe('2');
  });

  it('can decrement count on button click', () => {
    let app = mount(<Counter />);
    let button = app.find('button#decrement');
    expect(app.state('counter')).toBe(0);

    button.simulate('click');
    expect(app.state('counter')).toBe(-1);
    expect(app.find('h4#counter').text()).toBe('-1');

    button.simulate('click');
    expect(app.state('counter')).toBe(-2);
    expect(app.find('h4#counter').text()).toBe('-2');
  });

  it('renders consistently', () => {
    const tree = renderer.create(<Counter />).toJSON();

    expect(tree).toMatchSnapshot();
  })



})