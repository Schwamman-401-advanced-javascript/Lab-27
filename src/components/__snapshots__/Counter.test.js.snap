// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> renders consistently 1`] = `
<div>
  <h4
    className=""
    id="counter"
  >
    0
  </h4>
  <button
    id="increment"
    onClick={[Function]}
  >
    Increment
  </button>
  <button
    id="decrement"
    onClick={[Function]}
  >
    Decrement
  </button>
</div>
`;
